api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class CartController < ::Spree::Api::V2::BaseController
+          def add_item
+            variant = Spree::Variant.find(params[:variant_id])
+
+            spree_authorize! :update, current_order, order_token
+            spree_authorize! :show, variant
+
+            dependencies[:add_item_to_cart].call(order: current_order, variant: variant, quantity: params[:quantity])
+            render json: serialized_current_order, status: 201
+          end
+
+          def remove_item
+            variant = Spree::Variant.find(params[:variant_id])
+
+            spree_authorize! :update, current_order, order_token
+            spree_authorize! :show, variant
+
+            dependencies[:remove_item_from_cart].new.call(order: current_order, variant: variant, quantity: params[:quantity])
+
+            render json: serialized_current_order, status: 200
+          end
+
+          def set_item_quantity
+            variant = Spree::Variant.find(params[:variant_id])
+            line_item = Spree::LineItem.find_by(order: current_order, variant: variant)
```

Comment:
`line_item = Spree::LineItem.find_by(order: current_order, variant: variant)` 

-> 

`line_item = current_order.line_items.find_by!(variant: variant)`

easier to read + will raise 404 rather than 50x